JS-Algorithms/challenges.js

```diff
@@ -70,7 +98,22 @@ The last 3 digits for the sum of powers from 1 to 10 is "317"
 ***** */
 
 const ownPower = (number, lastDigits) => {
-  // YOUR CODE HERE...
+  const power = (base, exponent) => {
+    let result = 1n;
+    for (let i = 0n; i < exponent; i++) {
+      result *= base;
+    }
+    return result;
+  };
+
+  const ArrayNumbers = Array.from({ length: number }, (_, i) =>
+    power(BigInt(i + 1), BigInt(i + 1))
+  );
+
+  const NumberSum = ArrayNumbers.reduce((acc, elm) => acc + elm);
```

Comment:
Array.from({ length: number }, (_, i) => ...) is creating an array of size number, which is fine, but using a simple for loop will be more readable and performant for this task, especially when dealing with large numbers.

Reply:
We can performed the sum within the same for loop where we are calculating the power